src/main/java/org/jboss/modules/Main.java

```diff
@@ -248,6 +252,10 @@ public static void main(String[] args) throws Throwable {
                         }
                         defaultSecMgr = true;
                     } else if ("-secmgrmodule".equals(arg)) {
+                        if (!SECURITY_MANAGER_SUPPORTED) {
+                            System.err.printf("The security manager is not supported in Java %d.", Runtime.version().feature());
+                            System.exit(1);
+                        }
```

Comment:
Since these checks are repeated, you could instead just check below e.g. `if (actualVersion >= 24 && (secMgrModule != null || defaultSegMgr)) { .. print the error .. }`.

Reply:
That makes sense.